src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -17,15 +18,15 @@ public void spoonMavenLauncherTest() {
 		// without the tests
 		MavenLauncher launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
 
-		assertEquals(23, launcher.getEnvironment().getSourceClasspath().length);
+		assertEquals(189, launcher.getEnvironment().getSourceClasspath().length);
 
 		// 56 because of the sub folders of src/main/java
-		assertEquals(58, launcher.getModelBuilder().getInputSources().size());
+		assertEquals(59, launcher.getModelBuilder().getInputSources().size());
 
 		// with the tests
 		launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
 
-		assertEquals(30, launcher.getEnvironment().getSourceClasspath().length);
+		assertEquals(194, launcher.getEnvironment().getSourceClasspath().length);
```

Comment:
Could you add a new test to check this new behaviour: you did this PR because on some projects dependency resolution was not working. So could you put pom.xml files that wasn't working previously and that is working now, as a test?

Reply:
It's not that it used to not work on certain poms but more than some types were not resolved because not existing in the classpath where they could have been if MavenLauncher generated a classpath more similar to Maven.
I am unsure it is testable just with a pom. The problem appear when analyzing an AST. 
I could add a test that check that for a maven project (may be spoon itself), all type are resolvable, even those defined in dependencies. WDYT?

Reply:
> I could add a test that check that for a maven project (may be spoon itself), all types are resolvable, even those defined in dependencies. WDYT?

Sounds good yes.

Reply:
Done.